### ISSUE TITLE
The platforms flag should be single-valued

### ISSUE THREAD

**Issue by katre** [bazelbuild/bazel#19807]
### Description of the feature request:

Right now, the `--platforms` flag accepts multiple values but only the first is actually used.

Instead, the flag should be single-valued and give an error if multiple values are passed.

Further, the flag should actually be `--platform` and uses should be migrated, but that may be harder to manage.

### Which category does this issue belong to?

Configurability

### What underlying problem are you trying to solve with this feature?

Poor decision decisions at the start of the platforms project

**Comment by katre** [comment]
@sdtwigg We discussed this recently

**Comment by chaselatta** [comment]
I put together a quick repro case which shows the issue we discussed. https://github.com/chaselatta/platform_bug_repro

**Comment by fmeum** [comment]
I remember a design doc by @brentleyjones that (if I recall correctly) gave meaning to a multi-valued `--platforms` flag: A target is built for the first platform in the list that it is `target_compatible_with`, which allows for pretty clean flagless multi-platform builds. I am noting this here to make sure that if we migrate, we are very sure that we aren't going to migrate back eventually.

**Comment by brentleyjones** [comment]
Yes, I would like to confirm if this means we will never adopt something like my proposal. And if so, if there are any future plans at all to support multi-platform builds in a nice way?

**Comment by katre** [comment]
@brentleyjones: Your proposal was very interesting, and we may decide to take it up in the future. At this point, however, I think the best approach is the one from https://github.com/bazelbuild/proposals/blob/main/designs/2023-06-08-standard-platform-transitions.md, where the controls are in BUILD files and Starlark rules.

The reason to handle this now are to reduce confusion, both for Bazel users (who frequently ask about what happens when they pass multiple values to `--platforms`) and for Bazel developers (who keep having to add defensive code around `PlatformOptions.platforms`).

If we decide to implement either @brentleyjones's proposal or something similar, we can change the flag semantics again to match whatever we decide to implement.

**Comment by katre** [comment]
Copying parts of my comment from #6519:

Steps for cleaning up `--platforms`:
1. Make the existing multi-valued flag single-valued, but leave the name.
2. Actually rename the flag to `--platform` (flag migrations are hard, sorry), leaving `--platforms` as a deprecated legacy name.
3. Actually remove `--platforms`.

**Comment by fmeum** [comment]
@katre Would step 1 include making it single-valued within transition implementation functions as well? I could see that being very breaking.

Could we keep `//command_line_option:platforms` list-valued until it is removed and introduce `//command_line_option:platform` as the separate single-valued alternative?

**Comment by katre** [comment]
Yes, that's part of why we're not moving on this faster.

Having two different flags also seems confusing. When we pick this up we'll consider which is less bad.

I consider multi-valued `--platforms` as one of the two biggest mistakes when starting the platforms project (the other was a separate `toolchains` parameter to rule instead of a new `attr.toolchain` attribute type).

**Comment by katre** [comment]
We don't have any news, or any plans to prioritize renaming the flag.

If you want multiplatform builds, you should write a custom transition to correctly encode the logic you need, because there's no consensus on a global multiplatform strategy for all use cases.
